pkg/asset/manifests/dns.go

```diff
@@ -229,6 +225,36 @@ func (d *DNS) Generate(ctx context.Context, dependencies asset.Parents) error {
 	return nil
 }
 
+// GetGCPPrivateZoneName attempts to find the name of the private zone for GCP installs. When a shared vpc install
+// occurs, a precreated zone may be used. If a zone is found (in this instance), then the zone should be paired with
+// the network that is supplied through the install config (when applicable).
+func GetGCPPrivateZoneName(ctx context.Context, client *icgcp.Client, installConfig *installconfig.InstallConfig, clusterID string) (string, error) {
+	privateZoneID := fmt.Sprintf("%s-private-zone", clusterID)
+	if installConfig.Config.GCP.NetworkProjectID != "" {
+		zone, err := client.GetDNSZone(ctx, installConfig.Config.GCP.ProjectID, installConfig.Config.ClusterDomain(), false)
+		if err != nil {
+			return "", fmt.Errorf("failed to get private zone for %q: %w", installConfig.Config.BaseDomain, err)
+		}
+		if zone != nil {
```

Comment:
So the `GetDNSZone` call can return `nil` even if there weren't any errors?

Reply:
Yes, it returns nil when there are no zones found. It is not an error to find no zones. This is an indicator that a zone must be created.